domain/src/main/java/org/jboss/set/mjolnir/archive/domain/UserRemoval.java

```diff
@@ -25,13 +25,16 @@
 @NamedQueries({
         @NamedQuery(name = UserRemoval.FIND_REMOVALS_TO_PROCESS,
                 query = "SELECT r FROM UserRemoval r WHERE r.started IS NULL" +
-                        " AND (remove_on <= CURRENT_DATE or remove_on IS NULL)")
+                        " AND (remove_on <= CURRENT_DATE or remove_on IS NULL)"),
+        @NamedQuery(name = UserRemoval.FIND_FINISHED_REMOVALS,
+                query = "SELECT r FROM UserRemoval r WHERE r.started > :jobStart AND r.status != 'STARTED' ORDER BY r.status DESC, r.username")
```

Comment:
Thank you, I added the condition, but I think we still got the same problem with those which fails in different interval, then the one in which they started. In those cases, there are no records about completion time or failure time.

Reply:
What if we just report only removals, completed in given interval, and all existing failures no matter when they occur, as we can assume that they will be restarted/retried in next interval.

Reply:
Back to the "(started > :since OR completed > :since)" - the completion time should be logged also in case of failure. But I guess we can't rely on that completely, it could happen that the batch job terminates without us being able to catch the exception and the completion time would not be logged...

I'm OK with filtering by completion time only and reporting failures. We will probably need additional way to detect removals hanging in STARTED state for a long time.